NtupleProducer/plugins/HTauTauNtuplizer.cc

```diff
@@ -4000,50 +4012,55 @@ void HTauTauNtuplizer::endRun(edm::Run const& iRun, edm::EventSetup const& iSetu
   
   std::regex reg;
   if (uncertScheme.find("MadGraph") != std::string::npos) {
-	reg = "<weight.+ PDF=\"(.*)\" .+>.*<\\/weight>";
+    reg = "<weight.+ PDF=\"(.*)\" .+>.*<\\/weight>";
   }
   else if (uncertScheme.find("Powheg") != std::string::npos) {
-	reg = "<weight.+> lhapdf=(.*) .*<\\/weight>";
+    reg = "<weight.+> lhapdf=(.*) .*<\\/weight>";
   }
   
   for (headers_const_iterator iter = run->headers_begin(); iter != run->headers_end(); iter++) {
-	std::vector<std::string> lines = iter->lines();
-	for (unsigned int iLine = 0; iLine < lines.size(); iLine++) {
-	  std::smatch match;
-	  std::regex_search(lines.at(iLine), match, reg);
-	  if(DEBUG) {
-		std::cout << lines.at(iLine) << std::endl;
-	  }
-	  
-	  if (match.size()==2) {
-		if (weight_lines_counter == 0) {
-		  pdf_id_qcd_scale_ref = std::atoi(match[1].str().c_str());
-		}
-		else if (weight_lines_counter == _MC_pdf_first_idx) {
-		  pdf_id_pdf_ref = std::atoi(match[1].str().c_str());
-		}
-		weight_lines_counter += 1;
-		if(DEBUG) {
-		  std::cout << weight_lines_counter << " " << pdf_id_qcd_scale_ref << " " << pdf_id_pdf_ref << std::endl;
-		}
-	  }
-	  // no other checks, should exit
-	  if (weight_lines_counter > _MC_pdf_first_idx) {
-		break;
-	  }
-	}
+    std::vector<std::string> lines = iter->lines();
+    for (unsigned int iLine = 0; iLine < lines.size(); iLine++) {
+      std::smatch match;
+      std::regex_search(lines.at(iLine), match, reg);
+      if(DEBUG) {
+        std::cout << lines.at(iLine) << std::endl;
+      }
+
+      if (match.size()==2) {
+        if (weight_lines_counter == _MC_QCDscale_indices[0]) {
+          pdf_id_qcd_scale_ref = std::atoi(match[1].str().c_str());
+        }
+        else if (weight_lines_counter == _MC_pdf_first_idx) {
+          pdf_id_pdf_ref = std::atoi(match[1].str().c_str());
+        }
+        if(DEBUG) {
+          std::cout << weight_lines_counter << " " << pdf_id_qcd_scale_ref << " " << pdf_id_pdf_ref << std::endl;
+        }
+      }
+      if (lines.at(iLine).find("<weight ") != std::string::npos) {
+        weight_lines_counter++;
+      }
+      // no other checks, should exit
+      if (weight_lines_counter > _MC_pdf_first_idx) {
+      break;
+      }
+    }
   }
 
   if (pdf_id_qcd_scale_ref != pdf_id_pdf_ref) {
-	throw cms::Exception("InvalidOption")
-	  << "PDF ID of QCD scale reference ("
-	  << pdf_id_qcd_scale_ref
-	  << ") weight does not match PDF ID of PDF reference ("
-	  << pdf_id_pdf_ref
-	  << ") weight";
+    throw cms::Exception("InvalidOption")
+      << "PDF ID of QCD scale reference ("
+      << pdf_id_qcd_scale_ref
+      << ") weight does not match PDF ID of PDF reference ("
+      << pdf_id_pdf_ref
+      << ") weight";
   }
 }
```

Comment:
Is there any conditions to check whether the passed uncertainty scheme is correct? We have an assert in the Python code, but not in here, and the future could bring a mismatch in case we forget to update the Python ```assert SCHEME in ...``` condition.